### ISSUE TITLE
Labels in datasets list filter

### ISSUE THREAD

**Issue by nitrosx** [SciCatProject/frontend#1570]
## Issue Name
Labels in datasets list filter

### Summary
When the latest PR for the Search Ui has been merged, the FE show "default label" instead of the correct filter label in the "filters and conditions" form in the datasets list page

### Steps to Reproduce
Start the latest FE from master and  load the datasets list page

### Current Behaviour
![image](https://github.com/user-attachments/assets/f57382b4-695f-415d-aaa4-e761cb13e9a5)

### Expected Behaviour
The expected behaviour is to have each label matching the dataset property the associated filter works on.

**Comment by Ingvord** [comment]
@nitrosx Thanks for reporting!

However, I can not reproduce it:

![Screenshot_20240829_152918](https://github.com/user-attachments/assets/930ad44f-54e1-41b9-a609-d61e5f362147)


All labels are displayed correctly!

Could you please provide more details on how to reproduce? Thanks

**Comment by nitrosx** [comment]
Let me do some more digging. Thanks for checking

**Comment by Ingvord** [comment]
Yeah, in the first case code uses `this.constructor.name` to resolve actual filter label, while in the later it uses string value from the state. The first one, I suspect, maybe platform dependent... See #1571 for details

**Comment by Junjiequan** [comment]
During the Docker image building process, all the code is minified, which results in class names being altered or shortened. However, the current implementation directly uses the class name to retrieve the corresponding filter label from a hardcoded mapping (labelMap). This approach fails in production because the minified class names do not match the original keys in the labelMap.

**Comment by Junjiequan** [comment]
I suggest hardcoding the labels directly in the HTML and removing labelMappings altogether. Based on the current structure, it appears that each filter has its own corresponding HTML, and it doesn’t seem likely that we will allow users to modify the label names.

**Comment by Ingvord** [comment]
There is another place in filter-settings, where we can not easily access actual instance of the component, see https://github.com/SciCatProject/frontend/blob/7d5457b540c58048e681541e3214206515b30514/src/app/datasets/datasets-filter/settings/datasets-filter-settings.component.html#L15

Platform dependent code has been removed in #1571

**Comment by Junjiequan** [comment]
In that case, I suggest storing the labelsMapping in the frontend.config.json instead of hardcoding it in the frontend. Additionally, we don’t really need the default-filters.config.json in the backend, as it clearly belongs in the frontend configuration. I realize it’s a bit late to comment on the default-filters.config.json after it has been merged, but I didn’t have the full picture of how everything works together at that time.

**Comment by Ingvord** [comment]
I am confused. How you would do then  https://github.com/SciCatProject/scicat-backend-next/issues/604 ?

**Comment by Junjiequan** [comment]
If you’re asking about the feasibility of the approach I mentioned above, we have some examples on the backend that show how to return values from frontend.config.json for the settings/default endpoint:

import config from "../config/frontend.config.json";
import theme from "../config/frontend.theme.json";
```
@Injectable()
export class AdminService {
  async getConfig(): Promise<Record<string, unknown> | null> {
    return config;
  }
}
```
The retrieved config values can be stored in the userSettings state or another appropriate place and used by different components on the frontend.

This is just my idea—it’s doable, but I’m not saying it’s the optimal way.

**Comment by Ingvord** [comment]
Well, I am confused even more than...

That is exactly how default filter settings work now, see e.g. https://github.com/SciCatProject/scicat-backend-next/blob/master/src/users/interceptors/default-user-settings.interceptor.ts

Let's start from the beginning  - what problem are you trying to solve here?

**Comment by nitrosx** [comment]
@Ingvord, let me jump in and try to clarify.
The BE should be completely agnostic about the settings for the FE. It should only offer the functionalities to serve the FE configuration based on the site installation and store the user settings as they are.

With your approach, FE and BE are sharing the hard coded filters config in code, which create a limiting dependency. FE config updates would require changes to both the FE and BE. It will also be hard to decouple FE config from BE and where it can be stored.

Regarding the defaults, we think that it should be the task of the FE, through its configuration, to set them. 
The FE requests the user settings to the BE. Once it receives the data, the FE checks if the relevant settings are present. If they are not, it adds the relevant defaults from its configuration. This approach concentrate all the relevant logic in the FE and limits the interaction between FE and BE to the API defined.

While reviewing your PR, @Junjiequan and myself discussed extensively and, thanks to your work, found some issues due to the original requirements. Instead of wasting everybody's time in lengthy discussions, we think it is more productive to provide our suggestions through a PR directly on this PR/branch, so we can have a more focused discussion.

**Comment by Ingvord** [comment]
Resolved via #1571
